calling-apis/post-requests-with-axios.md

```diff
@@ -0,0 +1,221 @@
+
+<!-- prettier-ignore-start -->
+```js
+{
+    firstName: 'Fred',
+    lastName: 'Flintstone'
+}
+```
+<!-- prettier-ignore-end -->
+
+Matteo sees that there are two keys, `firstName` and `lastName`. From his knowledge of APIs, these keys don't look like they're describing parts of an HTTP request like HTTP headers or params. Additionally, this example request is being made to `/user`.
+
+Matteo hypothesizes that this object literal holds the HTTP request body data, and he can test his hypothesis against his API.
+
+### Sending a Request to the Ada Trip Reserving Service™ Trip API™™™
+
+Matteo is ready to try to make a `POST` request to the back-end service. He [determines from the API documentation](https://github.com/AdaGold/trip-api) that the request he wants to make is:
+
+- `POST` request to `https://trektravel.herokuapp.com/trips`
+- Request Body:
+  - `name`
+  - `continent`
+  - `about`
+  - `category`
+  - `weeks`
+  - `cost`
+
+The `post` function takes in two parameters: URL, and an optional object. This optional object represents the HTTP request body data that we want to send.
+
+Matteo could write this code:
+
+```js
+const axios = require("axios");
+
+const tripData = {
+  name: "Matteo's Chill Trip to Chicago",
+  continent: "North America",
+  about: "A tour around good architecture and hot dogs.",
+  category: "casual",
+  weeks: 1,
+  cost: 180,
+};
+
+axios
+  .post("https://trektravel.herokuapp.com/trips", tripData)
+  .then((response) => {
+    console.log("response:", response);
+    console.log("response data:", response.data);
+  })
+  .catch((error) => {
+    console.log("error:", error);
+    console.log("error response:", error.response);
+  })
+  .finally(() => {
+    console.log("finally done!");
+  });
+```
+
+With this code, Matteo can begin to make more `POST` requests in JavaScript!
+
+#### Refactoring: Inline `tripData`
+
+<details style="max-width: 700px; margin: auto;">
+
+  <summary>
+    Let's refactor our code! Instead of initializing the tripData variable, let's pass in an object literal.
+  </summary>
+
+```js
+const axios = require("axios");
+
+axios.post("https://trektravel.herokuapp.com/trips", {
+  name: "Matteo's Chill Trip to Chicago",
+  continent: "North America",
+  about: "A tour around good architecture and hot dogs.",
+  category: "casual",
+  weeks: 1,
+  cost: 180,
+});
+//   proceed as normal...
+//   .then((response) => { })
+//   .catch((error) => { })
+//   .finally(() => { });
+```
+
+</details>
+
+## Verifying Error Handling
+
+Matteo wants to confirm that his code can handle unsuccessful HTTP requests.
+
+To confirm this, he will temporarily purposely make an API call with bad trip data.
+
+He refactors his code to use a new object, `badTripData`, which has a blank `name`. According to [the API documentation](<(https://github.com/AdaGold/trip-api)>), it should not be a successful call.
```

Comment:
to confirm this sounds repetitive with previous paragraph
this is NOT a refactor
```suggestion
To do so, he will temporarily make an API call with bad trip data on purpose.

He modifies his code to use a new object, `badTripData`, which has a blank `name`. According to [the API documentation](<(https://github.com/AdaGold/trip-api)>), it should not be a successful call.
```